src/misc/unicode.cpp

```diff
@@ -229,6 +418,17 @@ void Grapheme::AddMark(const uint16_t in_code_point)
 	std::sort(marks_sorted.begin(), marks_sorted.end());
 }
 
+void Grapheme::CopyMarksFrom(const Grapheme& other)
+{
+	if (!is_valid || !other.is_valid) {
+		// Can't add combining mark to/from invalid grapheme
+		return;
+	}
+
+	marks        = other.marks;
+	marks_sorted = other.marks_sorted;
+}
```

Comment:
Because the if doesn't circumvent multiple code paths, suggest testing for the positive scenario and moving the assignments inside.

Reply:
OK, changed.